transform/transformers/survey_transformer.py

```diff
@@ -0,0 +1,70 @@
+import json
+import logging
+import os
+
+from structlog import wrap_logger
+
+from transform.settings import SDX_FTP_IMAGE_PATH, SDX_FTP_DATA_PATH, SDX_FTP_RECEIPT_PATH, SDX_RESPONSE_JSON_PATH
+from transform.transformers import ImageTransformer
+from transform.transformers.survey import Survey
+from transform.utilities.formatter import Formatter
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+class SurveyTransformer:
+    """Superclass for specific survey transformers.
+
+    Common functionality for transformer classes.
+    Subclasses must provide their own implementations for create_pck() and create_receipt().
+
+    """
+
+    def __init__(self, response, sequence_no):
+        self.response = response
+        self.sequence_no = sequence_no
+        self.logger = logger
+        self.ids = Survey.identifiers(response, seq_nr=sequence_no)
+        self.survey = Survey.load_survey(self.ids)
+        self.image_transformer = ImageTransformer(self.logger, self.survey, self.response,
+                                                  sequence_no=self.sequence_no, base_image_path=SDX_FTP_IMAGE_PATH)
+
+    def create_pck(self):
```

Comment:
If the base version of this and `create_receipt` do basically nothing then I think it should raise an exception (`raise NotImplementedError` I think) if you try to invoke it without implementing your own.  This'll show the intent a lot more clearly and ensure proper usage

Reply:
Not all surveys will need to override it (e.g. EPE survey will not be creating a pck). Even so it might be more readable if in those cases they explicitly return None - what do you think?

Reply:
that's a really good point... until the EPE unique snowflake, everything always would've wanted to implement it.  If there are legit ones like you have here then leaving this as is is fine.  Might be worth putting a comment on these functions saying you really should implement your own, but there are rare occasions where not implementing it is fine.  I think signalling intent is important cause at a glance, every transformer should implement it.

Reply:
Gone for fancy abstract methods - EPE can explicitly return None